WordPress/Classes/Services/MediaServiceRemoteCoreREST.swift

```diff
@@ -0,0 +1,217 @@
+import Foundation
+import Combine
+import WordPressCore
+import WordPressShared
+import WordPressAPI
+import WordPressAPIInternal
+import WordPressKit
+
+/// A `MediaServiceRemote` implementation that uses the WordPress core REST API (`/wp-json/wp/v2/media`).
+class MediaServiceRemoteCoreREST: NSObject, MediaServiceRemote {
```

Comment:
(nit) For future refactoring/integration, I'd suggest doing it the other way around: adding a `wordpress-rs` protocols with `wordpress-rs` types and adding conformance to `WordPressKit` so that the new code wouldn't depend on it.

Reply:
> adding a wordpress-rs protocols with wordpress-rs types

I don't think we need to do this part, because the heavy-lifting of implementing REST API is already done in wordpress-rs.

> so that the new code wouldn't depend on it.

I don't think there would be much new code directly depending on this `class MediaServiceRemoteCoreREST: MediaServiceRemote`, because the reason this class exists is to create an equivalent of `MediaServiceRemoteXMLRPC` and `MediaServiceRemoteREST`, so that it can be placed at where those types are called. In the future, when we are ready to remove WordPressKit completely, I imagine we'll just copy `RemoteMedia` over and delete `MediaServiceRemote`, which would be minimal changes.